### PR TITLE
add logger usage and logger tests consistently to services - master

### PR DESCRIPTION
**Pull request by clemensutschig**
@renedupont - can you please check as well, especially the snyk part

### CODE REVIEW

File: src/org/ods/services/JenkinsService.groovy [hunk 1]
```diff
@@ -23,10 +23,9 @@ class JenkinsService {
         }
 
         logger.debug "Collecting (applicable) testresults from location: '${xUnitResultDir}'"
-        def logCommand = logger.debugMode ? '' : 'set +x'
         script.sh(
             script: """
-                ${logCommand}
+                ${logger.shellScriptDebugFlag}
```
Comment (michaelsauter) [inline src/org/ods/services/JenkinsService.groovy:28]: Hmm, this is a bit confusing for me. Everywhere else, debugging is opt-in, whereas here it is opt-out. Not all places use `${logger.shellScriptDebugFlag}`, so most have `-x` (the Jenkins default), and only the few places here set `+x` when debugging is off.

I guess much more effort would be needed to have this consistent everywhere (e.g. by making `PipelineSteps` aware of this and modifying the passed `script`).

I'd probably leave this like it is unless we go all-in (which I would not do right now).

File: src/org/ods/services/OpenShiftService.groovy [hunk 2]
```diff
@@ -301,7 +301,7 @@ class OpenShiftService {
 
     int startBuild(String project, String name, String dir) {
         steps.sh(
-            script: "oc -n ${project} start-build ${name} --from-dir ${dir}",
+            script: "oc -n ${project} start-build ${name} --from-dir ${dir} ${logger.OCPDebugFlag}",
```
Comment (michaelsauter) [inline src/org/ods/services/OpenShiftService.groovy:304]: Love the addition. As a nitpick, could we use `logger.ocDebugFlag`? I think that would better reflect what the flag is for (the `oc` binary) and avoids `OCPD` all uppercase ;)

File: src/org/ods/services/SonarQubeService.groovy [hunk 3]
```diff
@@ -58,6 +61,7 @@ class SonarQubeService {
             script.sh(
                 label: 'Generate CNES Report',
                 script: """
+                ${logger.OCPDebugFlag}
```
Comment (michaelsauter) [inline src/org/ods/services/SonarQubeService.groovy:64]: This looks odd - is this a mistake?

File: src/org/ods/component/ScanWithSnykStage.groovy [hunk 4]
```diff
@@ -56,7 +56,8 @@ class ScanWithSnykStage extends Stage {
             "buildFile=${config.buildFile}, " +
             "failOnVulnerabilities=${config.failOnVulnerabilities}, " +
             "severityThreshold=${config.severityThreshold}, " +
-            "additionalFlags=${config.additionalFlags ? config.additionalFlags.toListString() : ''}."
+            "additionalFlags=${config.additionalFlags ? config.additionalFlags.toListString() : ''}." +
+            "${logger.debugMode ? '-d' : ''}"
```
Comment (michaelsauter) [inline src/org/ods/component/ScanWithSnykStage.groovy:60]: Personally I think we don't need the extra info about the debug mode here ...

Comment (renedupont) [inline src/org/ods/component/ScanWithSnykStage.groovy:60]: I assume what you actually wanted to have is:

```
        if (logger.debugMode) {
            config.additionalFlags += '-d'
        }

        logger.info 'Scanning for vulnerabilities with ' +
            "organisation=${config.organisation}, " +
            "projectName=${config.projectName}, " +
            "buildFile=${config.buildFile}, " +
            "failOnVulnerabilities=${config.failOnVulnerabilities}, " +
            "severityThreshold=${config.severityThreshold}, " +
            "additionalFlags=${config.additionalFlags ? config.additionalFlags.toListString() : ''}."
```

File: src/org/ods/services/JenkinsService.groovy (hunk 1 above)
Comment (clemensutschig) [reply src/org/ods/services/JenkinsService.groovy:28]: @michaelsauter  confused... what do you want, or not :)?

File: src/org/ods/services/OpenShiftService.groovy (hunk 2 above)
Comment (clemensutschig) [reply src/org/ods/services/OpenShiftService.groovy:304]: nitpick, thats right ... I'll change that

File: src/org/ods/services/SonarQubeService.groovy (hunk 3 above)
Comment (clemensutschig) [reply src/org/ods/services/SonarQubeService.groovy:64]: ahh. great catch!

File: src/org/ods/component/ScanWithSnykStage.groovy (hunk 4 above)
Comment (clemensutschig) [reply src/org/ods/component/ScanWithSnykStage.groovy:60]: I think it's worthwhile, especially when calling external jars / tools

Comment (renedupont) [reply src/org/ods/component/ScanWithSnykStage.groovy]: sorry, I just tested it, it needs to be:
```
        if (logger.debugMode) {
            if (config.additionalFlags) {
                config.additionalFlags += '-d'
            } else {
                config.additionalFlags = ['-d']
            }
        }
```

**Review by renedupont — APPROVED** [review]
snyk part looks good now

**Review by michaelsauter — APPROVED** [review]

**Pull request merged**